ArdusatSDK.h

```diff
@@ -49,51 +69,12 @@ typedef enum {
 } data_unit_t;
 
 /**
- * Data types for cells in data structures.
- */
-// schema for these constants is 			0000XXLL XX for type, LL for len-1
-#define DATA_CELLTYPE_HEX	    0x00
-#define DATA_CELLTYPE_HEX8		0x00	//	00000000
-#define DATA_CELLTYPE_HEX16		0x01	//	00000001
-#define DATA_CELLTYPE_HEX24		0x02	//	00000010
-#define DATA_CELLTYPE_HEX32		0x03	//	00000011
-
-#define DATA_CELLTYPE_INT	    0x04
-#define DATA_CELLTYPE_INT8		0x04	//	00000100
-#define DATA_CELLTYPE_INT16		0x05	//	00000101
-#define DATA_CELLTYPE_INT24		0x06	//	00000110
-#define DATA_CELLTYPE_INT32		0x07	//	00000111
-
-#define DATA_CELLTYPE_UINT	  0x08
-#define DATA_CELLTYPE_UINT8		0x08	//	00001000
-#define DATA_CELLTYPE_UINT16	0x09	//	00001001
-#define DATA_CELLTYPE_UINT24	0x0A	//	00001010
-#define DATA_CELLTYPE_UINT32	0x0B	//	00001011
-
-//#define DATA_CELLTYPE_STR		0x0D	//	00001101 EXCEPTION TO THE SCHEMA
-#define DATA_CELLTYPE_FLOAT		0x0F	//  00001111
-
-// utility constants
-#define DATA_CELLTYPE_LONGINT	  0x07	// (INT4)
-#define DATA_CELLTYPE_SHORTINT	0x05	// (INT2)
-#define DATA_CELLTYPE_INTEGER	  0x05	// (SHORTINT) (INT2)
-#define DATA_CELLTYPE_ULONGINT	0x0B	// (UINT4)
-#define DATA_CELLTYPE_USHORTINT	0x09	// (UINT2)
-#define DATA_CELLTYPE_BYTE		  0x00	// (HEX1)
-
-
-/**
  * The data header contains generic information about the data record.
  */
-#define SENSORDATA_HEADER_VERSION	 1
 struct _data_header_v1 {
-	//uint8_t version;		// version of the data packet (for compatibility mgt)
-	//uint8_t length;			// len of the data packet (256 is enough ?)
-	//uint8_t dimensionality;	// size of the cell array
-	//uint8_t celltype;		// type of the cell (0 = byte, 1 = float, etc…)
-	uint8_t unit;			// unit (standard) of the values (e.g. meter, m/s^2, etc.)
-	uint8_t sensor_id;		// id of the sensor that generated this data
-	uint32_t timestamp;		// millis for timestamping the data
+	char unit;			// unit (standard) of the values (e.g. meter, m/s^2, etc.)
+	char sensor_id;		// id of the sensor that generated this data
```

Comment:
these should be `unsigned char`, not `char`. *Technically* it is undefined (by the c standard) whether a `char` will be signed or unsigned - it should be specified if you're using it as a number, instead of a character.